rust/src/delta.rs

```diff
@@ -512,6 +513,21 @@ impl DeltaTable {
         }
     }
 
+    /// Create a new [`DeltaTable`] from a [`DeltaTableState`] without loading any
+    /// data from backing storage.
+    ///
+    /// NOTE: This is for advanced users. If you don't know why you need to use this method,
+    /// please call one of the `open_table` helper methods instead.
+    pub fn new_with_state(storage: ObjectStoreRef, state: DeltaTableState) -> Self {
```

Comment:
Could we make this `pub crate` too?
```suggestion
    pub crate fn new_with_state(storage: ObjectStoreRef, state: DeltaTableState) -> Self {
```